### ISSUE TITLE
Attention mask is ignored when using `scaled_dot_product` with flash attention cuda enabled

### ISSUE THREAD

**Issue by FSSRepo** [pytorch/pytorch#122753]
### 🐛 Describe the bug

In the code, `attn_mask` is not passed when using flash attention with cuda and cdnn:

https://github.com/pytorch/pytorch/blob/c81c9ba472ca46d5acac94ea7eea4dbd264a93fb/aten/src/ATen/native/transformers/attention.cpp#L664-L690

### Versions

Latest version
